libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/SchemaHelper.kt

```diff
@@ -38,7 +36,29 @@ internal object SchemaHelper {
     return clientBuilder.build()
   }
 
-  internal fun executeQuery(map: Map<String, Any?>, url: String, headers: Map<String, String>, insecure: Boolean): Response {
+  internal fun executeQuery(
+      query: Query<*>,
+      endpoint: String,
+      headers: Map<String, String>,
+      insecure: Boolean,
+  ): String {
+    val composer = DefaultHttpRequestComposer(endpoint)
+    val apolloRequest = ApolloRequest.Builder(query)
+        .httpHeaders(headers.map { HttpHeader(it.key, it.value) })
+        .build()
+    val httpRequest = composer.compose(apolloRequest)
+    val httpEngine = DefaultHttpEngine(newOkHttpClient(insecure))
+    val httpResponse = runBlocking { httpEngine.execute(httpRequest) }
+    val bodyStr = httpResponse.body?.use {
+      it.readUtf8()
+    }
+    check(httpResponse.statusCode in 200..299 && bodyStr != null) {
+      "Cannot get schema from $endpoint: ${httpResponse.statusCode}:\n${bodyStr ?: "(empty body)"}"
+    }
+    return bodyStr
```

Comment:
Parse `bodyStr` here to make sure the server response is valid?

Reply:
It's done upstream in `SchemaDownload.download()`, but I agree it makes sense to do it here 👍

Reply:
Good point! Used the generated parser in the last commit.